grpc/tests/src/test/gas_counter.rs

```diff
@@ -0,0 +1,236 @@
+                                                                  * the heap */
+        // Grow by N pages
+        (
+            Some(opcode_costs.op_const),
+            Instruction::I32Const(GROW_PAGES as i32),
+        ),
+        // memory.grow is metered by the number of pages
+        (
+            Some(opcode_costs.grow_memory * (GROW_PAGES + 1)),
+            Instruction::GrowMemory(0),
+        ),
+        (Some(opcode_costs.op_const), Instruction::I32Const(0)),
+        (Some(opcode_costs.store), Instruction::I32Store(0, 4)), /* Store `grow_memory` result
+                                                                  * whatever it is */
+        // Sentinel instruction that's required to be present but it's not accounted for
+        (None, Instruction::End),
+    ];
+
+    let instructions = opcodes.iter().map(|(_, instr)| instr.clone()).collect();
+    let accounted_opcodes: Vec<_> = opcodes.iter().filter_map(|(cost, _)| *cost).collect();
+
+    let session_bytes = make_session_code_with(instructions);
+
+    let exec_request = {
+        // NOTE: We use computed "do nothing" WASM module because it turns out "do_nothing" in
+        // AssemblyScript actually does "nop" which really "does something": (func (;10;)
+        // (type 4)   nop)
+
+        let do_nothing_bytes = make_minimal_do_nothing();
+
+        let deploy_item = DeployItemBuilder::new()
+            .with_address(*DEFAULT_ACCOUNT_ADDR)
+            .with_session_bytes(do_nothing_bytes, RuntimeArgs::default())
+            .with_empty_payment_bytes(runtime_args! {
+                ARG_AMOUNT => *DEFAULT_PAYMENT
+            })
+            .with_authorization_keys(&[*DEFAULT_ACCOUNT_ADDR])
+            .with_deploy_hash([43; 32])
+            .build();
+        ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+    };
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+
+    builder.run_genesis(&DEFAULT_RUN_GENESIS_REQUEST);
+
+    let payment_cost = {
+        let mut forked_builder = builder.clone();
+        forked_builder.exec(exec_request).commit().expect_success();
+        forked_builder.last_exec_gas_cost()
+    };
+
+    let exec_request = {
+        let deploy_item = DeployItemBuilder::new()
+            .with_address(*DEFAULT_ACCOUNT_ADDR)
+            .with_session_bytes(session_bytes, RuntimeArgs::default())
+            .with_empty_payment_bytes(runtime_args! {
+                ARG_AMOUNT => *DEFAULT_PAYMENT
+            })
+            .with_authorization_keys(&[*DEFAULT_ACCOUNT_ADDR])
+            .with_deploy_hash([42; 32])
+            .build();
+        ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+    };
+
+    builder.exec(exec_request).commit().expect_success();
+
+    let gas_cost = builder.last_exec_gas_cost() - payment_cost;
+    let expected_cost = accounted_opcodes.clone().into_iter().map(Gas::from).sum();
```

Comment:
```suggestion
    let expected_cost = Gas::from(accounted_opcodes.iter().sum::<u32>());
```
If we do this, we can avoid adding `impl Sum for Gas` to gas.rs.

Reply:
That's valid point though my intention here was to make a sum where partial values are Gas - similar to what EE would do during execution